### PR TITLE
Update Understand-determine-basal.md

### PR DESCRIPTION
**Pull request by parenthetic**
- Mostly updating iob.json documentation
- Minor additional edits

### CODE REVIEW

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Understanding the oref0-determine-basal logic
+# Understanding the *oref0-determine-basal.js* logic
```
Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: This is better as-is I think. NodeJS installs it as oref0-determine-basal.

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md [hunk 2]
```diff
@@ -28,24 +28,24 @@ The determine-basal algorithm requires a number of inputs, which are passed in J
...
-  * netbasalinsulin = net of basal insulin compared to preprogrammed pump basal rate
+  * iob = Units of Insulin on Board (IOB), ***net*** of your pre-programmed basal rates. Net IOB takes all pre-programmed basal, OpenAPS temp basal, and bolus insulin into account. Note: `iob` can be negative when OpenAPS temp basal rate is below your pre-programmed basal rate (referred to as "low-temping").
+  * activity = Units of insulin active in the previous minute. Approximately equal to (net IOB, 1 minute ago) - (net IOB, now).
+  * bolussnooze = Units of bolus IOB, if duration of insulin activity (dia) was half what you specified in your pump settings. (`dia_bolussnooze_divisor` in profile.json is set by default to equal 2, but you may adjust this if you'd like OpenAPS to activate a low-temp sooner or later after bolusing.) `bolussnooze` is used in *oref0-determine-basal.js* to determine how long to avoid low-temping after a bolus while waiting for carbs to kick in.
+  * basaliob = Insulin on board attributed to basal rates (pre-programmed and OpenAPS temp basal rates), excluding the IOB effect of boluses
```
Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: basaliob is also net, so it doesn't include the effect of scheduled basals.

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md (hunk 1 above)
Comment (parenthetic) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: Can you tell me a little bit more about how/where NodeJS installs code (or point me to a resource)? 

My thinking is that most folks wanting to understand the code will read this documentation first, and then potentially go to `oref0/bin/oref0-determine-basal.js` (which will ultimately lead them to `oref0/lib/determine-basal/determine-basal.js`). At least that's the path I followed. 

If NPM/NodeJS ultimately installs the code at a lower level than this, I think that would likely be beyond most users comprehension. I'm thinking about it as a sort of hierarchy: Docs > `bin/oref0-determine-basal.js` & `lib/determine-basal/determine-basal.js` > Ultimate NodeJS install. 

But maybe my mental model of it all isn't quite right...

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md (hunk 2 above)
Comment (parenthetic) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: My bad. I'll correct/update/improve the descriptions, but likely won't get to it until the weekend...

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md (hunk 1 above)
Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: Take a look at package.json - it details how those files are installed. The resulting command-line executable is just `oref0-determine-basal` without any path or suffix.

Comment (parenthetic) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: So `oref0-determine-basal` is essentially an alias within oref0 that calls `./bin/oref0-determine-basal.js`, which in turn calls `./lib/determine-basal/determine-basal.js`?

Is that the right way to think about it?

Still seems to me like the core logic of it all is located in `./lib/determine-basal/determine-basal.js` and that that might be worth at least hinting at in the docs so interested/curious readers/users can go there to actually read the code themselves. (Again, this is just based on my recollection of my path of discovery...)

Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: Take a read through bin/oref0-determine-basal.js: it's not just an alias, but rather the code that processes all the input files before calling the libs to do the bulk of the work.

Comment (parenthetic) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: I've read through `bin/oref0-determine-basal.js` and I get that it cleans up the input data before passing it all along to `lib/determine-basal/determine-basal.js`. (That's how I learned that script/program even existed.)

My "alias" reference was just to how package.json has the line "oref0-determine-basal": "./bin/oref0-determine-basal.js", which I loosely interpreted as an "alias." Or maybe my use of the term "calls" is what was confusing...

Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: Sounds like you understand it correctly then. If you'd like to add URL links to the files, that might help anyone trying to follow the bread crumbs?

Comment (parenthetic) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: I like that idea!
(Bread crumbs are useful...)

File: docs/docs/walkthrough/phase-3/Understand-determine-basal.md (hunk 2 above)
Comment (scottleibrand) [inline docs/docs/walkthrough/phase-3/Understand-determine-basal.md]: Looks like this one still needs fixed, then we may be ready to merge?

**Comment by danamlewis** [comment]
This LGTM. @scottleibrand did all of your flags get resolved? Merge away if so...

**Comment by parenthetic** [comment]
@danamlewis LGTM? What does that mean?

**Comment by danamlewis** [comment]
@parenthetic sorry for the acronym :) LGTM = looks good to me. SGTM = sounds good to me.

**Comment by parenthetic** [comment]
Yeah... just looked that up. I feel old.

**Comment by scottleibrand** [comment]
LGTM except for merge conflicts.

**Comment by danamlewis** [comment]
Resolved merge conflicts, merging. Thanks @parenthetic!

**Comment by parenthetic** [comment]
No problem.
Still more work to do here. (I see inconsistencies in what I've added,
even.)

I'll keep iterating as I find the time...

**Pull request merged**